core/src/main/java/com/klaytn/caver/abi/ABI.java

```diff
@@ -353,21 +338,81 @@ public static EventValues decodeLog(List<ContractIOType> inputs, String data, Li
         return new EventValues(indexedValues, nonIndexedValues);
     }
 
-//    private static int getLength(List<Type> parameters) {
-//        int count = 0;
-//        for (Type type:parameters) {
-//            if (type instanceof StaticArray) {
-//                count += ((StaticArray) type).getValue().size();
-//            } else {
-//                count++;
-//            }
-//        }
-//        return count;
-//    }
-
-//    private static boolean isDynamic(Type parameter) {
-//        return parameter instanceof DynamicBytes
-//                || parameter instanceof Utf8String
-//                || parameter instanceof DynamicArray;
-//    }
+    /**
+     * Decodes a function call data that composed of function selector and encoded input argument.
+     * <pre>Example :
+     * {@code
+     *  String encodedData = "0x24ee0.....";
+     *  String abi = "{\n" +
+     *                     "  \"name\":\"myMethod\",\n" +
+     *                     "  \"type\":\"function\",\n" +
+     *                     "  \"inputs\":[\n" +
+     *                     "    {\n" +
+     *                     "      \"type\":\"uint256\",\n" +
+     *                     "      \"name\":\"myNumber\"\n" +
+     *                     "    },\n" +
+     *                     "    {\n" +
+     *                     "      \"type\":\"string\",\n" +
+     *                     "      \"name\":\"mystring\"\n" +
+     *                     "    }\n" +
+     *                     "  ]\n" +
+     *                     "}";
+     *
+     * List<Type> params = caver.abi.decodeFunctionCall(abi, encoded);
+     * }
+     * </pre>
+     * @param functionAbi The abi json string of a function.
+     * @param encodedString The encode function call data string.
+     * @return List&lt;Type&gt;
+     * @throws ClassNotFoundException
+     */
+    public static List<Type> decodeFunctionCall(String functionAbi, String encodedString) throws ClassNotFoundException {
+        try {
+            ObjectMapper mapper = ObjectMapperFactory.getObjectMapper();
+            ContractMethod method = mapper.readValue(functionAbi, ContractMethod.class);
+            method.setSignature(ABI.encodeFunctionSignature(method));
+
+            if(method.getName() == null || method.getName().isEmpty() || method.getInputs() == null) {
```

Comment:
I think this should be checked before `ABI.encodeFunctionSignature(method)`.

Reply:
Thanks, I modified it.